### ISSUE TITLE
Character literals not highlighted

### ISSUE THREAD

**Issue by jtbandes** [kobalicek/reloaded-cpp#2]
It seems that single character literals such as `'x'` are not highlighted. This causes problems when `'"'` occurs, which instead of being a single character, actually starts a double quoted string. Instead the single quoted character should probably be scoped as `string.quoted.single`.

**Comment by Gruntfuggly** [comment]
If anybody is interested, I made this work by adding at line 533 of `c++.json`
```
        {
          "name": "string.quoted.single.cpp",
          "begin": "(u|u8|U|L)?'",
          "beginCaptures": {
            "0": { "name": "punctuation.definition.string.begin.cpp" },
            "1": { "name": "meta.encoding.cpp" }
          },
          "end": "'",
          "endCaptures": {
            "0": { "name": "punctuation.definition.string.end.cpp" }
          },
          "patterns": [
            { "name": "constant.character.escape.cpp", "match": "\\\\u\\h{4}|\\\\U\\h{8}" },
            { "name": "constant.character.escape.cpp", "match": "\\\\['\"?\\\\abfnrtv]" },
            { "name": "constant.character.escape.cpp", "match": "\\\\[0-7]{1,3}" },
            { "name": "constant.character.escape.cpp", "match": "\\\\x\\h+" },
            { "name": "constant.character.format.cpp", "match": "%(%|[-+ #0]*(\\*|\\d+)?(\\.(\\*|\\d+))?(hh|h|ll|l|L|j|z|t)?[diuoxXfFeEgGaAcspn])" }
          ]
        },
```

I'm sure this is a massive bodge, but it stops the highlighting breaking when you have `'"'` in your code. I know nothing about vscode/textmate grammars - it just replicates the double quoted string section with single quotes.

**Comment by kobalicek** [comment]
Guys sorry for a late reply, I have no problem to update the extension, but to be honest I was really waiting for something real not based on these insane rules :)

**Comment by kobalicek** [comment]
Should be fixed by f13ddbb24f1775e9cd92c5b415a1a618a4eb5a25
